### ISSUE TITLE
ProtonMail: could not properly parse signed message

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#3448]
[cannot-parse-signed.zip](https://github.com/FlowCrypt/flowcrypt-browser/files/6014847/cannot-parse-signed.zip)

With the above message I'm seeing the following:

![image](https://user-images.githubusercontent.com/6306961/108593940-abbfc280-736e-11eb-904f-6fbfd0a2ac22.png)

**Comment by rrrooommmaaa** [comment]
This is a signed and encrypted message sent in "PGP/inline" format by Protonmail with the sender's pubkey attached.

It consists of 1 text/plain MIME part (containing an `encryptedMsg` inline -- this is handled by `pgp_block` perfectly good)
and 2 attachments -- Public key of the sender and a signature (for this public key?).

So `replaceArmoredBlocks` is called first and it correctly processes the inline message and replaces the message body with the `pgp_block` eventually showing the green "encrypted" and "signed" badges.

Then `replaceAttachments` is called in second place, it finds a signature, wrongly assumes this is the signature for the text/plain part and wipes the previous `pgp_block` with a new one -- the new one treats the input as a text and a detached signature for it and fails to verify it.

Well, to fix this, a number of approaches may be applied depending on which use cases we want to cover.
For example, a sender may construct a text message containing samples like
```
a pgp-encrypted message looks like this
-----BEGIN PGP MESSAGE-----

...
-----END PGP MESSAGE-----
```
sign it and send it as a correct PGP/MIME message.

In this example we really shouldn't be looking inside the text, calling `MsgBlockParser`  ect, but do the straightforward PGP/MIME signed-only processing instead.

However, due to mail agents peculiarities we often get a malformed PGP/MIME message so looking for signatures in attachments also makes sense.

In the current issue, we should look into the message and ignore the signature (or rather apply the signature verification to the pubkey attachment. This is interesting -- as Protonmail keeps user keys on its servers it probably signs the attached keys with its own signature and we can try and automate verification of imported protonmail keys in a separate issue -- though I haven't found any docs on this). In the current issue we only have 
```
-----BEGIN PGP MESSAGE-----

...
-----END PGP MESSAGE-----
```
in the text body and nothing else -- this may help to diagnose the applied approach.

But generally, I see, the currently scattered logic between the web-content-script and a number of `pgp_block` iframes makes it hard to implement a simple and straightforward approach. Maybe after #4906 we should think about doing the whole decryption/verification process in the context of a web-content-script? The most gnarly problem is when we replace the visually displayed message with a number of `pgp_block` iframes, these iframes continue the processing each with its little piece of information, and if a new attachment pops up on the page, there is currently no way to pass this data to the iframes, all that the web-content-script can do is just wipe the previously created iframes and start over, and that's not very good too. 

@sosnovsky , @tomholub opinions?

**Comment by sosnovsky** [comment]
> In this example we really shouldn't be looking inside the text, calling MsgBlockParser etc, but do the straightforward PGP/MIME signed-only processing instead.

Agree, as valid pgp message should start with `BEGIN PGP...` without any text before. Just need to test if it won't break any existing functionality, but probably should be ok.

> In the current issue, we should look into the message and ignore the signature (or rather apply the signature verification to the pubkey attachment. 

Here pubkey and signature attachments have the same file name, just different extension. We can use this to check if signature is related to message or only to pubkey.

> Maybe after https://github.com/FlowCrypt/flowcrypt-browser/issues/4906 we should think about doing the whole decryption/verification process in the context of a web-content-script?

Sounds good, just need to check if `openpgp.js` will work well in content script. Then we can use it to optimize whole decryption process.

**Comment by tomholub** [comment]
> can try and automate verification of imported protonmail keys in a separate issue -- though I haven't found any docs on this)

anyway proton mail has wkd where the client always searches. so this is not needed, except for custom domains on proton mail. probably not worth

**Comment by rrrooommmaaa** [comment]
> Here pubkey and signature attachments have the same file name, just different extension. We can use this to check if signature is related to message or only to pubkey.

ok. We shouldn't hide the `sig` file in this case, should we?
![image](https://user-images.githubusercontent.com/23335944/222116630-57cfc795-be14-4681-b19e-7a184596dde1.png)

**Comment by tomholub** [comment]
I'd hide the sig file, if it's just signing another file which we don't have support for verifying in that manner, I'd not render it.

**Comment by rrrooommmaaa** [comment]
Hid the sig file
